File: Medium/Consecutive 1's not allowed/consecutive-1s-not-allowed.py
#User function Template for python3
class Solution:

	def countStrings(self,n):
	    ans=2
	    ans2=3
	    if(n==1):
	        return 2
	    if n==2:
	        return 3
	    for i in range(2,n):
	        ans3=(ans+ans2)%1000000007
	        ans=ans2
	        ans2=ans3
	    return ans3
	        
	        
    	# code here


#{ 
 # Driver Code Starts
#Initial Template for Python 3



# Driver code 
if __name__ == "__main__": 		
    tc=int(input())
    while tc > 0:
        n=int(input())
        ob = Solution()
        ans = ob.countStrings( n)
        print(ans)
        tc=tc-1
# } Driver Code Ends

File: Medium/Consecutive 1's not allowed/consecutive-1s-not-allowed.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class GFG {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine().trim());

            long ans = new Solution().countStrings(n);

            System.out.println(ans);
        }
    }
}
// } Driver Code Ends


// User function Template for Java

class Solution {
    long countStrings(int n) {
        // code here
        long f1=1;
        long f2=1;
        long f3=f1+f2;
        if (n==1){
            return f3;
        }
        for (int i=2;i<=n;i++){
            f1=f2 %(1000000007);
            f2=f3%(1000000007);
            f3=(f1+f2)%(1000000007);
        }
        return f3;
    }
}